hashicorp/consul-k8s
# Decoupling Consul's Role-Based Access Control (RBAC) from the core system.

### Changes proposed in this PR ###  
The goal of this requirement is to enhance the security architecture by decoupling Consul's Role-Based Access Control (RBAC) from the core system. This change is aimed at allowing Consul RBAC to be installed and configured as a prerequisite, ensuring that the permissions required to install the application and the cluster level RBAC can be managed independently. By implementing this solution, we will improve access management, reduce security risks, and streamline RBAC architecture.
**Goals:**

- Decoupling RBAC: Implement a mechanism that allows Consul RBAC to function independently of the Consul cluster, promoting a modular architecture.

- Minimizing Permissions: Ensure that users responsible for installing the RBAC and application operate with the least privilege principle, mitigating the risk posed by overly broad permissions.

- Improving Security Posture: Enhance the overall security of applications by limiting the attack surface exposed to potential threats.

**Benefits:**

- Enhanced Security: Reducing cluster-wide permissions decreases the risk of unauthorized access and potential vulnerabilities.

- Easier Compliance: By enabling granular access controls, organizations can more easily comply with security standards and regulations.

- Operational Efficiency: Streamlined RBAC permissions management leads to a more efficient development and deployment process, reducing friction between teams.

### How I've tested this PR ###

- Built the chart and deployed and tested with our Organization's product.
- Deployed Consul-RBAC as a separate chart maintained by us
- Tested  new RBAC components to validate individual functions and logic. This included  consul chart deployment along with other applications  and ensuring  normal functioning without any unexpected behaviors.

### How I expect reviewers to test this PR ###

- Deploy/Verify manifest of the consul chart with --set global.rbac.create=false value and ensure that this will not create any RBAC resources 
- Deploy with --set global.rbac.create=true value and ensure successful consul deployment along with RBAC resource creation.

### Checklist ###
